docs/source/architecture/transactions_and_batches.rst

```diff
@@ -0,0 +1,193 @@
+
+Transaction Data Structure
+==========================
+
+Transactions are serialized using Protocol Buffers.  They consists of two
+message types:
+
+.. literalinclude:: ../../../protos/transaction.proto
+   :language: protobuf
+   :caption: File: protos/transaction.proto
+   :linenos:
+
+Header, Signature, and Public Keys
+----------------------------------
+
+The Transaction header field is a serialized version of a TransactionHeader.
+The header is signed by the signer's private key (not sent with the
+transaction) and the resulting signature is stored in header_signature.  The
+header is present in the serialized form so that the exact bytes can be
+verified against the signature upon receipt of the Transaction.
+
+The verification process verifies that the key in signer_pubkey signed the
+header bytes resulting in header_signature.
+
+The batcher_pubkey field must match the public key used to sign the batch in
+which this transaction is contained.
+
+
+.. note::
+
+   The original header bytes as constructed from the sender are used
+   for verification of the signature.  It is not considered good practice to
+   de-serialize the header (for example, to a Python object) and then
+   re-serialize the header with the intent to produce the same byte sequence as
+   the original.  Serialization can be sensitive to programming language or
+   library, and any deviation would produce a sequence that would not match the
+   signature; thus, best practice is to alway use the original header bytes for
+   verification.
+
+Transaction Family
+------------------
+
+In Sawtooth Lake, the set of possible transactions are defined by an extensible
+system called transaction families.  Defining and implementing a new
+transaction family adds to the taxonomy of available transactions which can be
+applied. For example, in the :ref:`transaction-family-tutorial`, we
+define a transaction family called "xo" which defines a set of transactions for
+playing tic-tac-toe.
+
+In addition to the name of the transaction family (family_name), each
+transaction specifies a family version string (family_version).  The version
+string enables upgrading a transaction family while coordinating the nodes
+in the network to upgrade.  The payload_encoding field allows the transaction
+family to support multiple serialization mechanisms (for example, json or
+cbor).
+
+Dependencies and Input/Output Addresses
+---------------------------------------
+
+Transactions can depend upon other transactions, which is to say a dependent
+transaction cannot be applied prior to the transaction upon which it depends.
+
+The dependencies field of a transaction allows explicitly specifying the
+transactions which must be applied prior to the current transaction.  Explicit
+dependencies are useful in situations where transactions have dependencies but
+can not be placed in the same batch (for example, if the transactions are
+submitted at different times).
+
```

Comment:
In the absence of explicit dependencies, is there any relationship between the order in which the transactions appear in transaction_ids and the order in which they are executed?  I don't think so but it might be worth stating explicitly.